### PR TITLE
Add integration tests for Apache Pinot

### PR DESCRIPTION
**Pull request by mik-laj**
I wanted to update the [pinotdb](https://pypi.org/project/pinotdb/) library, so I added one integration test to be able to easily check this integration.

Part of: https://github.com/apache/airflow/issues/12120

Latest supported version for Airflow 2.0: 0.1.1 (2 years ago)
Latest supported version for Airflow 1.10: 0.1.1 (2 years ago) 
Latest version:  0.3.3 (5 months ago) 
Diff between releases date of latest version and latest supported version: 2 years  

```
root@4cc53fe49ba8:/opt/airflow# pytest tests/providers/apache/pinot/hooks/test_pinot.py -k Integrat -s
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.6.12, pytest-6.1.2, py-1.9.0, pluggy-0.13.1 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /opt/airflow, configfile: pytest.ini
plugins: forked-1.3.0, timeouts-1.2.1, cov-2.10.1, xdist-2.1.0, instafail-0.4.2, flaky-3.7.0, requests-mock-1.8.0, rerunfailures-9.1.1, celery-4.4.7
setup timeout: 0.0s, execution timeout: 0.0s, teardown timeout: 0.0s
collected 13 items / 12 deselected / 1 selected

tests/providers/apache/pinot/hooks/test_pinot.py::TestPinotDbApiHookIntegration::test_should_return_records ========================= AIRFLOW ==========================
Home of the user: /root
Airflow home /root/airflow
Skipping initializing of the DB as it was initialized already.
You can re-initialize the database by adding --with-db-init flag when running tests.
[2020-12-20 16:19:55,293] {base.py:74} INFO - Using connection to: id: pinot_broker_default. Host: pinot, Port: 8000, Schema: , Login: None, Password: None, extra: None
[2020-12-20 16:19:55,294] {pinot.py:275} INFO - Get the connection to pinot broker on pinot
PASSED

===================================================================================== 1 passed, 12 deselected in 1.78s =====================================================================================
```

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -181,7 +181,7 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'atlasclient>=0.1.2',
 ]
 azure = [
-    'azure-batch>=8.0.0',
+    'azure-batch>=8.0.0,<10.0.0',
```
Comment (potiuk) [inline setup.py]: You can likely remove that limitation as well.

Comment (mik-laj) [reply setup.py]: I had to add another restriction. [pyjwt](https://pypi.org/project/PyJWT/#history) was updated a few hours ago and is causing the problem.

**Comment by mik-laj** [comment]
We have next issue - mock<4.0.3 is not compatible with latest moto.  See: https://github.com/spulec/moto/issues/3535

**Comment by mik-laj** [comment]
@potiuk Finally green after pinning some additional libraries. Can you look at it?

**Comment by mik-laj** [comment]
@potiuk Rebased and CI are sad. Can you help me with it?

**Comment by potiuk** [comment]
> @potiuk Rebased and CI are sad. Can you help me with it?

Sure. Can you read the instructions below the error here:

https://github.com/apache/airflow/pull/13195/checks?check_run_id=1618826928

  msal 1.8.0 has requirement PyJWT[crypto]<2,>=1.0.0, but you have pyjwt 2.0.0.
  flask-jwt-extended 3.25.0 has requirement PyJWT<2.0,>=1.6.4, but you have pyjwt 2.0.0.
  ERROR: ^^^ Some dependencies are conflicting. See instructions below on how to deal with it. 

There are some instructions what to do, can you please follow it? I have some idea what to do but I would love to know if you reach the same conclusions and whether we can update the instructions to be cleaner? My goal is that anyone should be able to fix this kind of problems.

**Comment by mik-laj** [comment]
I pinned `pyjwt` to `<2`. https://github.com/apache/airflow/blob/6063583cb2f981880f0a75a97118b96ae401d9dd/setup.py#L396-L398
It helped until the new change https://github.com/apache/airflow/commit/82fa048c128b698a28d69d92e0c01d9b3953d883 was merged.  I have a feeling that this is not a problem with my change now, but it is a more general problem in CI setup.

**Comment by mik-laj** [comment]
@potiuk Looks like all problems are fixed.

File: setup.cfg [hunk 2]
```diff
@@ -113,11 +113,16 @@ install_requires =
...
+    # https://github.com/snowflakedb/snowflake-connector-python/blob/v2.3.6/setup.py#L201
+    pyOpenSSL<20.0.0
     pygments>=2.0.1, <3.0
+    # snowflake-connector-python, flask-jwt-extended, msal are not compatible with newest version.
+    pyjwt<2
     python-daemon>=2.1.1
     python-dateutil>=2.3, <3
     python-nvd3~=0.15.0
-    python-slugify>=3.0.0,<5.0
+    python-slugify>=3.0.0,<5.
```
Comment (mik-laj) [inline setup.cfg]:
```suggestion
    python-slugify>=3.0.0,<5.0
```

**Review by potiuk — APPROVED** [review]

**Pull request merged**